python/tk_multi_workfiles/browser_form.py

```diff
@@ -669,3 +675,50 @@ def _on_step_filter_changed(self, step_list):
         :param step_list: A list of Shotgun Step dictionaries.
         """
         self.step_filter_changed.emit(get_filter_from_filter_list(step_list))
+
+    # Task status filter methods
+    def __get_project_id(self, bundle):
+        """
+        Helper method to return the project id for the current context.
+
+        :param bundle: The current app bundle.
+        """
+        if bundle.tank.pipeline_configuration.is_site_configuration():
+            return None
+        else:
+            return bundle.tank.pipeline_configuration.get_project_id()
+
+    def populate_task_status_list(self):
+        """
+        Populate the task status list from the project schema.
+        """
+        app = sgtk.platform.current_bundle()
+        project_id = self.__get_project_id(app)
+        if project_id:
+            sg = app.shotgun._sg
+            logger.debug(
+                f"Task Status Filter: getting statuses from project {project_id}"
+            )
+            task_status_list = sg.schema_field_read(
```

Comment:
You should be able to use the app `shotgun`  property directly to make api requests:
```suggestion
            task_status_list = app.shotgun.schema_field_read(
```